### PR TITLE
Create recipe for hpv-meta in version 1.0 for conda-forge.

### PR DESCRIPTION
**Pull request by faramir**
Dear @conda-forge/help-java I'm author and maintainer of `hpv-meta` and author of this recipe. The receipe is ready for review!

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (LICENSE file in source tarball file).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by faramir** [comment]
Dear @conda-forge/help-java I'm author and maintainer of `hpv-meta` and author of this recipe. The recipe is ready for review!

**Comment by faramir** [comment]
@conda-forge/help-java I have prepared version for Linux and OSX ready for review.

Building for windows as the build "here" shows strange error, that you can see on the message from [30ef91c](https://github.com/conda-forge/staged-recipes/pull/28498/commits/30ef91c2509a9efeb348e746374721b37b517941) and here:
```2024-12-10T13:30:34.4305589Z source tree in: D:\bld\hpv-kite_1733837381864\work
2024-12-10T13:30:34.4570076Z WARNING: Using legacy MSVC compiler setup.  This will be removed in conda-build 4.0. If this recipe does not use a compiler, this message is safe to ignore.  Otherwise, use {{compiler('<language>')}} jinja2 in requirements/build.
2024-12-10T13:30:34.4618143Z Rewriting env in output: {'BUILD_PREFIX': 'D:\\bld\\hpv-kite_1733837381864\\_h_env',
2024-12-10T13:30:34.4618765Z  'PREFIX': 'D:\\bld\\hpv-kite_1733837381864\\_h_env',
2024-12-10T13:30:34.4619419Z  'SRC_DIR': 'D:\\bld\\hpv-kite_1733837381864\\work'}
2024-12-10T13:30:34.4845728Z
2024-12-10T13:30:36.5095991Z Traceback (most recent call last):
2024-12-10T13:30:36.5097207Z   File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 2510, in build
2024-12-10T13:30:36.5104630Z     windows.build(
2024-12-10T13:30:36.5105161Z   File "D:\Miniforge\Lib\site-packages\conda_build\windows.py", line 339, in build
2024-12-10T13:30:36.5107033Z     check_call_env(
2024-12-10T13:30:36.5107563Z   File "D:\Miniforge\Lib\site-packages\conda_build\utils.py", line 406, in check_call_env
2024-12-10T13:30:36.5109222Z     return _func_defaulting_env_to_os_environ("call", *popenargs, **kwargs)
2024-12-10T13:30:36.5111826Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-12-10T13:30:36.5112326Z   File "D:\Miniforge\Lib\site-packages\conda_build\utils.py", line 382, in _func_defaulting_env_to_os_environ
2024-12-10T13:30:36.5113959Z     raise subprocess.CalledProcessError(proc.returncode, _args)
2024-12-10T13:30:36.5115313Z subprocess.CalledProcessError: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 255.
2024-12-10T13:30:36.5115702Z
2024-12-10T13:30:36.5116199Z The above exception was the direct cause of the following exception:
2024-12-10T13:30:36.5116620Z
2024-12-10T13:30:36.5117101Z Traceback (most recent call last):
2024-12-10T13:30:36.5117547Z   File "D:\a\1\s\.ci_support\build_all.py", line 356, in <module>
2024-12-10T13:30:36.5118087Z     build_all(os.path.join(root_dir, "recipes"), args.arch)
2024-12-10T13:30:36.5118711Z   File "D:\a\1\s\.ci_support\build_all.py", line 185, in build_all
2024-12-10T13:30:36.5120484Z     build_folders(recipes_dir, folders, arch, channel_urls)
2024-12-10T13:30:36.5120976Z   File "D:\a\1\s\.ci_support\build_all.py", line 248, in build_folders
2024-12-10T13:30:36.5122955Z     conda_build.api.build([recipe], config=get_config(arch, channel_urls))
2024-12-10T13:30:36.5123461Z   File "D:\Miniforge\Lib\site-packages\conda_build\api.py", line 211, in build
2024-12-10T13:30:36.5125243Z     return build_tree(
2024-12-10T13:30:36.5125789Z            ^^^^^^^^^^^
2024-12-10T13:30:36.5126304Z   File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 3653, in build_tree
2024-12-10T13:30:36.5135985Z     packages_from_this = build(
2024-12-10T13:30:36.5137049Z                          ^^^^^^
2024-12-10T13:30:36.5137508Z   File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 2514, in build
2024-12-10T13:30:36.5145380Z     raise BuildScriptException(str(exc), caused_by=exc) from exc
2024-12-10T13:30:36.5146483Z conda_build.exceptions.BuildScriptException: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 255.
2024-12-10T13:30:43.3289605Z ##[error]Cmd.exe exited with code '1'.
2024-12-10T13:30:43.3784004Z ##[section]Finishing: Build recipes
```

What is wrong and how to fix this? I checked that the recipe can be successfully executed using `conda-build` on `miniforge3`.

**Comment by faramir** [comment]
@conda-forge-admin could you help?

**Comment by faramir** [comment]
@conda-forge-admin, please ping team

**Comment by faramir** [comment]
OK, 
@conda-forge-admin, please ping team conda-forge/help-java

**Comment by faramir** [comment]
Sorry, forgotten @
@conda-forge-admin, please ping team @conda-forge/help-java

File: recipes/hpv-kite/meta.yaml
Comment (ocefpaf) [inline recipes/hpv-kite/meta.yaml:32]: @xhochy I'm not comfortable reviewing a Java recipe but this one looks OK. Do you mind taking a second look before we merge?

File: recipes/hpv-kite/meta.yaml
Comment (danielnachun) [inline recipes/hpv-kite/meta.yaml:12]:
```suggestion
build:
  noarch: generic
```

Comment (danielnachun) [reply recipes/hpv-kite/meta.yaml:12]: This should be `noarch: generic`, and you can then remove the `bld.bat`.  The Windows failure can be ignored in that case, as the build will only be run on `linux-64` in the actual feedstock.

File: recipes/hpv-kite/build.sh
Comment (danielnachun) [inline recipes/hpv-kite/build.sh:13]: Putting `.jar` files in `${PREFIX}/bin` can cause weird issues later for users.  A better approach is to use `${PREFIX}/libexec/hpv-kite`.

So I'd suggest the following instead:
```
mkdir -p ${PREFIX}/libexec/hpv-kite
cp build/libs/*.jar ${PREFIX}/libexec/hpv-kite
```

Comment (danielnachun) [inline recipes/hpv-kite/build.sh:13]:
```suggestion
mkdir -p ${PREFIX}/libexec/hpv-kite
cp build/libs/*.jar ${PREFIX}/libexec/hpv-kite
```
Putting `.jar` files in `${PREFIX}/bin` can cause weird issues later for users.  A better approach is to use `${PREFIX}/libexec/hpv-kite`.

File: recipes/hpv-kite/build.sh
Comment (danielnachun) [inline recipes/hpv-kite/build.sh:5]:
```suggestion
sed -i 's/id "java"/id "java"\nid "com.github.jk1.dependency-license-report" version "latest.release"/' build.gradle
```
Please add this line so that the `dependency-license-report` plugin can be run.

File: recipes/hpv-kite/build.sh
Comment (danielnachun) [inline recipes/hpv-kite/build.sh:7]:
```suggestion
./gradlew clean build jar createDependenciesJar
./gradlew generateLicenseReport
```
We can run `generateLicenseReport` once we've used `sed` to add the `dependency-license-report` plugin.

File: recipes/hpv-kite/meta.yaml
Comment (danielnachun) [inline recipes/hpv-kite/meta.yaml:14]:
```suggestion
```
This won't be needed with `noarch: generic`.

File: recipes/hpv-kite/meta.yaml
Comment (danielnachun) [inline recipes/hpv-kite/meta.yaml:39]:
```suggestion
```
This is not needed now that we use SPDX identifiers.

File: recipes/hpv-kite/meta.yaml
Comment (danielnachun) [inline recipes/hpv-kite/meta.yaml:40]:
```suggestion
  license_file: 
    - LICENSE
    - build/reports/dependency-license/ 
```
This will automatically add the third-party licenses downloaded by the `dependency-license-report` plugin.

**Comment by danielnachun** [comment]
One last overall request - could you translate comments into English?   While I know what these steps do because I've submitted a lot of `gradle` recipes, it maybe hard for other users to follow what the build script does without translating the comments, and for better or worse English has become the standard language for open source software.  I'm happy to check spelling and grammar!

File: recipes/hpv-kite/meta.yaml
Comment (danielnachun) [inline recipes/hpv-kite/meta.yaml:18]:
```suggestion
    - openjdk >=17,<22
    - sed
```
For good measure let's make sure we are using GNU `sed` - BSD `sed` has issues with more advanced `sed` operations.

Also, I think we should just pin to `openjdk =21` here because that is newest LTS version your software supports.

**Comment by faramir** [comment]
As stated [here](https://github.com/conda-forge/staged-recipes/pull/28498#discussion_r1887301349) by @danielnachun 
> This should be `noarch: generic`, and you can then remove the `bld.bat`. The Windows failure can be ignored in that case, as the build will only be run on `linux-64` in the actual feedstock.

The win_64 built failed but it can be ignored.
I'm not sure if it is possible to modify the recipe so that the red cross will not be shown.

**Comment by danielnachun** [comment]
Right now there is nothing that can be done for the Windows "failure".  I'm hoping we can eventually change the way CI works to get rid of this, but for now simply notifying the reviewers the recipe is `noarch: generic` is all we can do.

@xhochy this should be good to merge now.

**Comment by ocefpaf** [comment]
@danielnachun thanks so much for the review!

@faramir Uwe is quite busy so, if he can't see this recipe in 1-2 days, ping me for a merge.

File: recipes/hpv-kite/meta.yaml
Comment (xhochy) [inline recipes/hpv-kite/meta.yaml:25]: This is already in run and the test section should also test whether `run` is correct.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Minor comment, otherwise LGTM

**Comment by faramir** [comment]
Thank you @danielnachun @xhochy and @ocefpaf for your feedback. I have made suggested updates and I think that the recipe is ready to merge now.

**Review by xhochy — APPROVED** [review]

**Pull request merged**
